src/config.cpp

```diff
@@ -59,31 +59,126 @@ void parse_uint32(void* value_ptr, const char* txt) {
+        }
+        line = trim(line);
+        // Check for section headers
+        if (line[0] == '[' && line[line.size() - 1] == ']') {
+            continue;
+        } else {
+            // Parse key-value pairs
+            size_t pos = line.find('=');
+            if (pos != std::string::npos) {
+                std::string key = trim(line.substr(0, pos));
+                std::string value = trim(line.substr(pos + 1));
+
+                // Store values (if any)
+                if (value != "") {
+                    umap[key] = value;
+                }
+            }
+        }
+    }
+    config_stream.close();
+}
+
 void parse_env() {
+    std::unordered_map<std::string, std::string> fileConfigValues;
+    read_config_file(fileConfigValues);
     for (auto& opt : gConfigOptions) {
         std::string var_name = "CLVK_";
         std::string optname_upper(opt.name);
         std::transform(optname_upper.begin(), optname_upper.end(),
                        optname_upper.begin(), ::toupper);
         var_name += optname_upper;
-        // printf("var_name = '%s' ", var_name.c_str());
         const char* txt = getenv(var_name.c_str());
         if (txt == nullptr) {
-            //    printf("is not set\n");
-            continue;
-        }
-        // printf("is set\n");
-        void* optval = const_cast<void*>(opt.value);
-        switch (opt.type) {
-        case config_option_type::string:
-            parse_string(optval, txt);
-            break;
-        case config_option_type::boolean:
-            parse_boolean(optval, txt);
-            break;
-        case config_option_type::uint32:
-            parse_uint32(optval, txt);
-            break;
+            if (fileConfigValues.find(opt.name) == fileConfigValues.end() ||
+                fileConfigValues[opt.name].length() == 0) {
+                continue;
+            }
+            auto curr_val = fileConfigValues[opt.name];
+            config_option_type optType = infer_type(fileConfigValues[opt.name]);
+            void* optval = const_cast<void*>(opt.value);
+            switch (optType) {
+            case config_option_type::string:
+                parse_string(optval, curr_val.c_str());
+                continue;
+            case config_option_type::boolean:
+                parse_boolean(optval, curr_val.c_str());
+                continue;
+            case config_option_type::uint32:
+                parse_uint32(optval, curr_val.c_str());
+                continue;
+            }
+        } else {
+            void* optval = const_cast<void*>(opt.value);
+            switch (opt.type) {
+            case config_option_type::string:
+                parse_string(optval, txt);
+                break;
+            case config_option_type::boolean:
+                parse_boolean(optval, txt);
+                break;
+            case config_option_type::uint32:
+                parse_uint32(optval, txt);
+                break;
+            }
```

Comment:
Parsing config options from a file should be done in a dedicated function called from `init_config` (see TODO) rather than mixing the logic with environment variable parsing. This split makes it easier to reason about how the various sources for configuration are layered on top of each other and to change the order if we want to.

Reply:
done thanks!